js/src/App.vue

```diff
@@ -21,17 +21,35 @@ export default {
       tournaments: [],
       // The main websocket object
       ws: null,
+      reconnections: 0,
       user: new User(),
     }
   },
   methods: {
+    reconnect: function () {
+      this.$delete("ws")
+
+      if (this.reconnections < 30) {
+        console.warn(`closed uncleanly, reconnecting (try number ${this.reconnections})`)
+        setTimeout(() => {
+          this.reconnections = this.reconnections + 1
+          this.connect()
+        }, 500)
```

Comment:
Right, I think this works just fine after all.

The loop goes like this. Let's say the server closed the connection: the `onclose` handler will run `reconnect()`. This will make us try to connect again after half a second. If the server is not there, we will fail, and the `onclose` handler will run again. This raises `this.reconnections`, so eventually we will reach 30, and stop trying. 

Alternatively, if the server comes back up, the loop is closed because the `onclose` handler will not run, and no reconnection will be scheduled.

Does that clear it up?